src/utils/groups.ts

```diff
@@ -89,3 +90,41 @@ export function choosePlatformIcon(platform?: string) {
   }
   return undefined
 }
+
+export function applyFilters(groups: Group[], filters: Filters): Group[] {
+  let newGroups: Group[] = groups
+  if (filters.year !== undefined) {
+    newGroups = newGroups.filter(group => {
+      return group.year === filters.year
+    })
+  }
+  if (filters.platform !== undefined) {
+    newGroups = newGroups.filter(group => {
+      return group.platform === filters.platform
+    })
+  }
+  if (filters.type !== undefined) {
+    newGroups = newGroups.filter(group => {
+      return group.type === filters.type
+    })
+  }
+  if (filters.course !== undefined) {
+    newGroups = newGroups.filter(group => {
+      return group.degree === filters.course
+    })
+  }
+  return newGroups
+}
+
+export function searchGroups(groups: Group[], search: string): Group[] {
+  return groups.filter(group => {
+    try {
+      if (group.class === null) {
+        return false
+      }
+      return group.class.toLowerCase().includes(search.trimEnd().toLowerCase())
+    } catch (err) {
+      console.log(err)
+    }
```

Comment:
sono stupido io o niente qui dentro può throware? non credo sto `try-catch` faccia niente ahah

Reply:
magari `group.class` potrebbe essere `undefined` invece che `null`?
ti converrebbe semplicemente scrivere
```ts
if (!group.class) return false
```
grazie javascript e i concetti completamente sensati btw ahah

Reply:
Ok right. Perché quando avevo implementato la funzione CompareBiYear per confrontare i campi "year" dei gruppi, avevo avuto delle difficoltà perchè dal server i campi venivano settati a null invece che undefined (campo vuoto) e non mi ricordo se `!a` con `a=null` restituisse lo stesso risultato che con `a=undefined`.

Non escludo qualche errore da parte mia però.

Comunque per questo caso sembra più logico come hai scritto, ad ogni modo il campo group.class dovrebbe essere sempre non nullo, il controllo l'ho fatto per avere una sicurezza in più

Reply:
Ok capito grazie. Magari prossimamente una volta che il il json per i gruppi è quello giusto, faccio pulizia nella logica dell'ordinamento dei gruppi e rimuovo anche tutti i controlli per i formati sbagliati.